src/main/java/htsjdk/samtools/CigarOperator.java

```diff
@@ -89,114 +117,116 @@
     /**
      * @param b CIGAR operator in character form as appears in a text CIGAR string
      * @return CigarOperator enum value corresponding to the given character.
+     * @deprecated use {@link #fromChar(int)} instead.
      */
+    @Deprecated
     public static CigarOperator characterToEnum(final int b) {
-        switch (b) {
-        case 'M':
-            return M;
-        case 'I':
-            return I;
-        case 'D':
-            return D;
-        case 'N':
-            return N;
-        case 'S':
-            return S;
-        case 'H':
-            return H;
-        case 'P':
-            return P;
-        case '=':
-            return EQ;
-        case 'X':
-            return X;
-        default:
-            throw new IllegalArgumentException("Unrecognized CigarOperator: " + b);
-        }
+        return fromChar(b);
+    }
+
+    public static CigarOperator fromChar(final int ch) {
+       if ((ch & ~0x7F) == 0) {
+           final CigarOperator result = byte2Value[ch & 0x7F];
+           if (result == null) {
+               throw new IllegalArgumentException("Unrecognized CigarOperator: " + ch);
+           }
+           return result;
+       } else {
+           throw new IllegalArgumentException("Unrecognized CigarOperator: " + ch);
+       }
     }
 
     /**
      * @param i CIGAR operator in binary form as appears in a BAMRecord.
      * @return CigarOperator enum value corresponding to the given int value.
+     * @deprecated use {@link #fromBamCode(int)}.
      */
+    @Deprecated
     public static CigarOperator binaryToEnum(final int i) {
-        switch(i) {
-            case OP_M:
-                return M;
-            case OP_I:
-                return I;
-            case OP_D:
-                return D;
-            case OP_N:
-                return N;
-            case OP_S:
-                return S;
-            case OP_H:
-                return H;
-            case OP_P:
-                return P;
-            case OP_EQ:
-                return EQ;
-            case OP_X:
-                return X;
-            default:
-                throw new IllegalArgumentException("Unrecognized CigarOperator: " + i);
+        return fromBamCode(i);
+    }
+
+    /**
+     * Looks-up the operator given its BAM code.
+     * <p>
+     *     It is guaranteed that {@code CigarOperator.fromBamCode(op.bamCode()) == op}.
+     * </p>
+     *
+     * @param bamCode the query BAM code.
+     * @return never {@code null}.
+     * @throws IllegalArgumentException if the input BAM code is not valid.
+     * @see #bamCode
+     */
+    public static CigarOperator fromBamCode(final int bamCode) {
```

Comment:
Is this actually faster than the switch statement?  It's slightly less verbose but it seems likely that it will be implemented pretty similarly as a lookup in a small precomputed table.

Reply:
will leave it as it is.